src/overlays/actors/ovl_En_Fr/z_en_fr.c

```diff
@@ -27,99 +139,966 @@ const ActorInit En_Fr_InitVars = {
+void EnFr_ContinueFrogSong(EnFr* this, GlobalContext* globalCtx) {
+    s32 counter;
+    EnFr* frog;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1BF10.s")
+    if (this->frogSongTimer == 0) {
+        EnFr_OcarinaMistake(this, globalCtx);
+    } else {
+        this->frogSongTimer--;
+        if (globalCtx->msgCtx.msgMode == 0x32) { // Frogs Song
+            counter = 0;
+            for (i = 0; i < ARRAY_COUNT(sEnFrPointers.frogs); i++) {
+                frog = sEnFrPointers.frogs[i];
+                if (frog != NULL && frog->actionFunc == EnFr_ChooseJumpFromLogSpot) {
+                    continue;
+                } else {
+                    counter++;
+                }
+            }
+            if (counter == 0 && CHECK_BTN_ALL(globalCtx->state.input[0].press.button, BTN_B)) {
+                EnFr_OcarinaMistake(this, globalCtx);
+                return;
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C05C.s")
+        if (globalCtx->msgCtx.msgMode == 0x33) {
+            globalCtx->msgCtx.msgMode = 0x31;
+            switch (globalCtx->msgCtx.unk_E410) {
+                case FROG_BTN_A:
+                    EnFr_SetupJumpingUp(this, FROG_BLUE);
+                    break;
+                case FROG_BTN_C_DOWN:
+                    EnFr_SetupJumpingUp(this, FROG_YELLOW);
+                    break;
+                case FROG_BTN_C_RIGHT:
+                    EnFr_SetupJumpingUp(this, FROG_RED);
+                    break;
+                case FROG_BTN_C_LEFT:
+                    EnFr_SetupJumpingUp(this, FROG_PURPLE);
+                    break;
+                case FROG_BTN_C_UP:
+                    EnFr_SetupJumpingUp(this, FROG_WHITE);
+            }
+            if (EnFr_IsFrogSongComplete(this, globalCtx)) {
+                this->actor.textId = 0x40AC;
+                EnFr_SetupReward(this, globalCtx, false);
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C18C.s")
+void EnFr_SetupReward(EnFr* this, GlobalContext* globalCtx, u8 unkCondition) {
+    EnFr_DeactivateButterfly();
+    if (unkCondition) {
+        func_80078884(NA_SE_SY_TRE_BOX_APPEAR);
+    } else {
+        func_80078884(NA_SE_SY_CORRECT_CHIME);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C1E4.s")
+    func_800ED858(0); // Comment from z_en_fu.c: seems to be related to setting instrument type (Disable ocarina)
+    globalCtx->msgCtx.msgMode = 0x37;
+    this->actionFunc = EnFr_PrintTextBox;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C248.s")
+void EnFr_PrintTextBox(EnFr* this, GlobalContext* globalCtx) {
+    func_8010B680(globalCtx, this->actor.textId, &this->actor); // Looking up text box message
+    this->actionFunc = EnFr_TalkBeforeReward;
```

Comment:
can omit comment. not sure what looking up means, i think its actually displaying it. but yea we can wait for message decomp